src/cron/syncSubscription.ts

```diff
@@ -0,0 +1,29 @@
+import { syncSubscription } from '../common';
+import { StorageKey, StorageTopic, generateStorageKey } from '../config';
+import { logger } from '../logger';
+import { getRedisListLength, popFromRedisList } from '../redis';
+import { Cron } from './cron';
+
+const redisKey = generateStorageKey(
+  StorageTopic.CIO,
+  StorageKey.Reporting,
+  'global',
+);
+
+const cron: Cron = {
+  name: 'sync-subscription',
+  handler: async (con) => {
+    logger.info('syncing subscriptions');
+
+    while ((await getRedisListLength(redisKey)) > 0) {
+      // Store it in a set to remove duplicates
+      const userIds = new Set(await popFromRedisList(redisKey, 100));
+
+      await syncSubscription([...userIds], con);
+
+      logger.info(`synced subscriptions for ${userIds.size} users`);
```

Comment:
```suggestion
      await syncSubscription([...userIds], con);
      await new Promise((resolve) => setTimeout(resolve, 1000));

      logger.info(`synced subscriptions for ${userIds.size} users`);
```

Probably could add this too, just to ensure we don't exceed the rate limit of 10 requests per second. In case the the list has >1000 entries.

Reply:
Yes, I imagine this can resolve quite fast, how many requests we send per `syncSubscription` call? @omBratteng

Reply:
Per `syncSubscription`, we do one call to them. so it is very much possible that we manage to do >10req/s.
But, that all depends on how many users are queued up. Could be 50, then we'd suffice with just one call, as it does up to 100 users.

Reply:
But then, better safe than sorry. Doesn't hurt to have it there. It's also a cron job, so doesn't affect execution time of any endpoints.

And if the while loop were to go haywire, it would run 100 times, taking 100 seconds, but that's fine I guess?

Reply:
Yes, I agree

Reply:
We can wait `100ms` , even if we executed support fast we would be at 10 req/s

Reply:
`200ms` just to be on the safe side?

Reply:
Added 200ms

Reply:
![measure](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExMmN2ZWhkMjVidTBvZzZlZ3h6dnk1b2gzbzAzc29tcWtoMmk3dXUxdiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/kEoe8FvrvXtFrBsu6J/giphy-downsized-large.gif)